feat: add email registration page

rayrunzewang opened the pull request:
feat: added user registration page

Added (dynamic) user registration page:
- Navigate to Sign up page;
- Fill up sign-up form 
  - Display input criteria as the user typing in each input field. 
  - Validate input value after click submit.
- Added unit tests for registration page and related components
- Sign-up form validation： 
- Password: 
  - Must contain 8-32 characters 
  - Must contain letter and number 
  - Must contain at least one special character 
- Display name: 
  - Must contain 4-15 characters 
  - Only letters, numbers, dashes and underscore are allowed
- Reference name: 
   - Currently we will have a pre-filled Reference Code 
   - We can define that in environment variables in both backend and frontend 
   - Is disabled field: currently this field in client app is disabled, user can’t edit it

Resolve CP-5

![2024-04-17 20_43_54-Register - BeeQuant](https://github.com/BeeQuantAI/platform_app/assets/80236130/e302147b-5e71-4628-9db1-a3f33d11a8d0)
![2024-04-17 20_44_25-Register - BeeQuant](https://github.com/BeeQuantAI/platform_app/assets/80236130/a8144841-ce87-4204-8046-a3e1c598b7a5)![2024-04-17 20_44_39-Register - BeeQuant](https://github.com/BeeQuantAI/platform_app/assets/80236130/54780b0b-989c-4ae5-82f5-cce8bd8c6d83)
![2024-04-17 20_46_02-Register - BeeQuant](https://github.com/BeeQuantAI/platform_app/assets/80236130/9f6f10e8-307f-4c75-a35e-a621deabd7c3)

Chris-Wang reviewed (changes requested):
Overall good, just some changes need to be made.

Also, please update description to add screenshot of your task, as per our PR standard

Chris-Wang reviewed (changes requested):
LGTM

please squash your commits and resolve conflict

Chris-Wang reviewed (changes requested):
Need squash commits

rayrunzewang committed:
feat: added user registration page

Added (dynamic) user registration page:
- Navigate to Sign up page;
- Fill up sign-up form
  - Display input criteria as the user typing in each input field.
  - Validate input value after click submit.
- Added unit tests for registration page and related components
- Sign-up form validation：
- Password:
  - Must contain 8-32 characters
  - Must contain letter and number
  - Must contain at least one special character
- Display name:
  - Must contain 4-15 characters
  - Only letters, numbers, dashes and underscore are allowed
- Reference name:
   - Currently we will have a pre-filled Reference Code
   - We can define that in environment variables in both backend and frontend
   - Is disabled field: currently this field in client app is disabled, user can’t edit it

Resolve CP-5

Chris-Wang reviewed (approved):
LGTM
